File: Search/SearchInsertPosition/SearchInsertPosition.py
class SearchInsertPosition:
    def searchInsertPosition(self, nums, k):
        if nums is none:
            return -1;
            
        start, end = 0, len(nums) - 1
        while start + 1 < end:
            mid = (start + end) / 2
            if nums[mid] == target:
                end = mid
            elif nums[mid] < target:
                start = mid
            else: 
                end = mid
        
        if nums[start] >= k:
            return start
        elif nums[end] >= k:
            return end
        else:
            return len(nums)   
         

File: Search/SearchInsertPosition/SearchInsertPosition.cc
class SearchInsertPosition {
	public:
		int searchInsertPosition(vecotr<int> nums, int k) {
			if(nums == null || nums.size == 0)
				return -1;
			
			int start = 0;
			int end = nums.size - 1;
			
			while(start + 1 < end) {
				int mid = (start + end) / 2;
				if(nums[start] > k) {
					start = mid;
				} if(nums[end] > k) {
					end = mid;					
			}
			return end;
		}
};